plugins/gitgrep.py

```
@@ -31,8 +34,35 @@ def grep(query: str, **params) -> ([str], [str]):
+            if attempt < MAX_RETRIES - 1:
+                time.sleep(RETRY_DELAY * (attempt + 1))
+                continue
+            return [], []
+        except requests.exceptions.JSONDecodeError:
+            last_error = "grep.app returned an empty or invalid response."
+            if attempt < MAX_RETRIES - 1:
+                time.sleep(RETRY_DELAY * (attempt + 1))
+                continue
+            return [], []
+        except requests.exceptions.RequestException as e:
+            last_error = f"Error reaching grep.app: {e}"
```

Flake8 (7.3.0)

[error] local variable 'last_error' is assigned to but never used

(F841)